ee/clickhouse/queries/funnels/funnel.py

```diff
@@ -69,8 +69,11 @@ def get_partition_cols(self, level_index: int, max_steps: int):
             if i < level_index:
                 cols.append(f"latest_{i}")
             else:
+                duplicate_event = 0
+                if i > 0 and self._filter.entities[i].id == self._filter.entities[i - 1].id:
```

Comment:
we're going to need a more complex condition for determining duplicates. Technically, if you have $pageview with some list of properties A, B, etc and another step with $pageview and properties X, Y, Z, they don't need to be treated as duplicates because they won't match the same

Reply:
Ah, interesting!

I also need to write tests with action events. Will add one for this too.

Can you point me to where we parse these properties for the query?

Reply:
Yup, it's actually right in the funnel base class. Look for `_build_step_query` and subsequently `_build_filters`

Reply:
Perfect, thanks!